### DISCUSSION TITLE
How to reach first parent of current node?

### DISCUSSION THREAD

**Discussion by Mirko-Tebaldi** [rectorphp/rector#7764] [Q&A] [answered]
# Question

I have some rows like this

    $time_max=strtotime(mysql_result($result,0,'data_max'));

Other like this

    $variable_name= call( call ( call ( strtotime(mysql_result($result,0,'data_max')) ) ) );

I'm creating a rule to refactore prev lines as 

    $date_string = mysql_result($result,0,'data_max');
    $time_max=strtotime($date_string);

and 

    $date_string = mysql_result($result,0,'data_max');
    $variable_name= call( call ( call ( strtotime($date_string) ) ) );

To do this I am creating a new node . Then I'd like to use NodesToAddCollector to add new node before the very first node of the current FuncCall I'am parsing.

Can I identifiy the 'first' node? the parent of parent of parent ... node of a current node ? 
Is there a method to simply return the "main" node , the 'full node' of 'current line' of code?

**Comment by samsonasik** [comment, marked as answer]
There is BetterNodeFinder->resolveCurrentStatement() for that, see https://github.com/rectorphp/rector-src/blob/74bb7a0a7153e3b4423dd816da23c5de55875fc2/src/PhpParser/Node/BetterNodeFinder.php#L495

**Reply by Mirko-Tebaldi** [reply]
How to use this? 

It retrieves a use PhpParser\Node\Stmt; 
It's not a Node

**Reply by samsonasik** [reply]
Stmt is a Node, you want the "main" of current Node, which is an Stmt.

If you just want to get parent, you can call `$node->getAttribute(AttributeKey::PARENT);

**Reply by samsonasik** [reply]
`BetterNodeFinder` is already inside rule, you can just use it, ref for example usage:

https://github.com/rectorphp/rector/blob/f913db606919b017aea8a79d4cbbc759e47cf3d8/rules/PSR4/Rector/FileWithoutNamespace/NormalizeNamespaceByPSR4ComposerAutoloadRector.php#L113

**Reply by Mirko-Tebaldi** [reply]
Yes this works. Thanks
